### ISSUE TITLE
Loss reweighting?

### ISSUE THREAD

**Issue by fcogidi** [VectorInstitute/cyclops#366]
In the PyTorch model wrapper class (`PTModel`) there is [a function](https://github.com/VectorInstitute/cyclops/blob/main/cyclops/models/wrappers/pt_model.py#L432) that was originally intended for re-weighting the loss for unbalanced datasets. This function is currently not implemented and can either be removed - because some PyTorch loss functions already support this - or refactored to a more general case for handling imbalanced datasets.
